gramps/gui/configure.py

```diff
@@ -1713,3 +1724,227 @@ def update_surn_height(self, obj, constant):
+                 ),
+                4, line_wrap=True)
+            for val in available_fonts:
+                if sel_font == val:
+                    break
+                active_val += 1
+            self.add_combo(self.grid,
+                _('Choose font'),
+                5, 'utf8.selected-font',
+                self.all_avail_fonts, callback=self.utf8_update_font, valueactive=True, setactive=active_val)
+            symbols = Symbols()
+            all_sbls = symbols.get_death_symbols()
+            all_symbols = []
+            for symbol in all_sbls:
+                all_symbols.append(symbol[1] + " " + symbol[0])
+            self.all_death_symbols = [x for x in enumerate(all_symbols)]
+            pos = config.get('utf8.death-symbol')
+            combo = self.add_combo(self.grid,
+                _('Select default death symbol'),
+                6, 'utf8.death-symbol',
+                self.all_death_symbols, callback=self.utf8_update_death_symbol, valueactive=True, setactive='')
+            combo.set_active(pos)
+            if config.get('utf8.selected-font') != "":
+                self.utf8_show_example()
+
+        return _('Genealogical Symbols'), self.grid
+
+    def can_we_use_genealogical_fonts(self, obj):
+        try:
+            import fontconfig
+            from gramps.gui.utils import ProgressMeter
+        except:
+            from gramps.gui.dialog import WarningDialog
+            WarningDialog(_("Cannot look for genealogical fonts"),
+                          _("I am not able to select genealogical fonts. "
+                            "Please, install the module fontconfig for python 3."),
+                          parent=self.uistate.window)
+            return False
+        fonts = fontconfig.query()
+        all_fonts = {}
+        symbols = Symbols()
+        nb_symbols = symbols.get_how_many_symbols()
+        self.in_progress = True
+        self.progress = ProgressMeter(_('Checking available genealogical fonts'),
+                                       can_cancel=True,
+                                       cancel_callback=self.stop_looking_for_font, parent=self.uistate.window)
+        self.progress.set_pass(_('Looking for all fonts with genealogical symbols.'), nb_symbols*len(fonts))
+        for idx in range(0, len(fonts)):
+            if not self.in_progress:
+                return
+            for rand in range(symbols.SYMBOL_MALE, symbols.SYMBOL_EXTINCT+1):
+                string = symbols.get_symbol_for_html(rand)
+                value = symbols.get_symbol_for_string(rand)
+                font = fonts[idx]
+                fontname = font.family[0][1]
+                try:
+                    vals = all_fonts[fontname]
+                except:
+                    all_fonts[fontname] = []
+                if font.has_char(value):
+                    if value not in all_fonts[fontname]:
+                        all_fonts[fontname].append(value)
```

Comment:
Since you don't use the order of the results here, this seems like a good place to use a `set` instead of a list.

Reply:
If I use a set instead of a list, I get : 
    if value not in all_fonts[fontname]:
TypeError: 'set' object is not subscriptable
perhaps there is a better way to do this.

Reply:
I think he meant to use a set instead of a list as the stored value in all_fonts[fontname].  If you use the defaultdict of his earlier suggestion, all_fonts gets initialized like
    all_fonts = defaultdict(set)  # you must use "from collections import defaultdict" at top of file
and then your try and 'if' gets replaced with
    if font.has_char(value):
         if value not in all_fonts[fontname]:
                all_fonts[fontname].add(value)

Reply:
I didn't use defaultdict. This was the problem.
Done

Reply:
Yes, that's what I meant. In fact, you don't need the `if` and can do `all_fonts[fontname].add(value)` unilaterally because the set will keep things unique. You should also remove [this try](https://github.com/gramps-project/gramps/pull/598/files#diff-8c90436ad9f31535f4cc5d1cd05e1a87R1832) because it doesn't do anything with `defaultdict`.

Reply:
Done